src/e2e-test/java/io/cdap/plugin/bigquerymultitable/stepdesign/BQMultiTableValidation.java

```diff
@@ -0,0 +1,231 @@
+
+        switch (columnType) {
+
+          case Types.INTEGER:
+            Integer sourceInteger = rsSource.getInt(currentColumnCount);
+            Integer targetInteger = bigQueryData.get(jsonObjectIdx).get(columnName).getAsInt();
+            Assert.assertEquals("Different values found for column : %s", sourceInteger, targetInteger);
+            break;
+          case Types.VARCHAR:
+            String sourceString = rsSource.getString(currentColumnCount);
+//                        String sour = bigQueryData.get(jsonObjectIdx).get(columnTypeName).getAsString();
+//                        String sou2 = bigQueryData.get(jsonObjectIdx).get(columnType)
+            String targetString = bigQueryData.get(jsonObjectIdx).get(columnName).getAsString();
+            Assert.assertEquals("Different values found for column : %s", sourceString, targetString);
+            break;
+
+        }
+        currentColumnCount++;
+      }
+      jsonObjectIdx++;
+    }
+    return true;
+  }
+
+  /**
+   * Retrieves the data from a specified BigQuery table and populates it into the provided list of objects.
+   *
+   * @param table        The name of the BigQuery table to fetch data from.
+   * @param bigQueryRows The list to store the fetched BigQuery data.
+   */
+
+  private static void getBigQueryTableData(String table, List<Object> bigQueryRows)
+    throws IOException, InterruptedException {
+
+    String projectId = PluginPropertyUtils.pluginProp("projectId");
+    String dataset = PluginPropertyUtils.pluginProp("dataset");
+    String selectQuery = "SELECT TO_JSON(t) FROM `" + projectId + "." + dataset + "." + table + "` AS t";
+    TableResult result = BigQueryClient.getQueryResult(selectQuery);
+    result.iterateAll().forEach(value -> bigQueryRows.add(value.get(0).getValue()));
+  }
+
+  /**
+   * Fetches the list of table names from the configured dataset in BigQuery.
+   *
+   * @return A TableResult object containing the table names.
+   * @throws IOException If an I/O error occurs.
+   * @throws InterruptedException If the current thread is interrupted while waiting.
+   */
+
+  public static TableResult getTableNamesFromDataSet() throws IOException, InterruptedException {
+    String projectId = PluginPropertyUtils.pluginProp("projectId");
+    String dataset = PluginPropertyUtils.pluginProp("dataset");
+    String selectQuery = "SELECT table_name FROM `" + projectId + "." + dataset + "`.INFORMATION_SCHEMA.TABLES ";
+
+    return BigQueryClient.getQueryResult(selectQuery);
+  }
+
+  /**
+   * Fetches the names of target tables from the dataset and returns the list of matching table names.
+   *
+   * @return A list of target table names from the dataset.
+   * @throws IOException If an I/O error occurs.
+   * @throws InterruptedException If the current thread is interrupted while waiting.
+   */
+
+  public static List<String> getTableByName() throws IOException, InterruptedException {
```

Comment:
change the method name to getTableNameFromMySQL()

Reply:
change done.